### PR TITLE
fix export Enum and custom datatype variables

### PR DESCRIPTION
**Pull request by bitkeeper**
fix for #352

### CODE REVIEW

File: opcua/common/xmlexporter.py [hunk 1]
```diff
@@ -228,7 +229,26 @@ def add_variable_common(self, node, el):
...
+        Returns:
+            NodeId of datatype base or None in case base datype can not be determined
+        """
+        if datatype.NamespaceIndex == 0 and type(datatype) == ua.NumericNodeId and datatype.Identifier <= 30:
+            return datatype
+        dtype_supers_nodes = get_node_supertypes(self.server.get_node(datatype), includeitself=True, skipbase=True)
+        dtype_supers_nodeids = [node.nodeid for node in dtype_supers_nodes if node.nodeid.NamespaceIndex == 0 and  node.nodeid.Identifier <= 30]
+        if len(dtype_supers_nodeids) >= 1:
+            return dtype_supers_nodeids[0]
+        return None
```
Comment (oroulet) [inline opcua/common/xmlexporter.py]: Did you copy the method from ua_utils.py? Can't you call it from ua_utils.py? you can modify/split the one in ua_utils.py if you need so

I think we should raise an exception if no data type is found, this is not normal. so raising an exception is correct and clearly shows there is an issue

File: tests/tests_xml.py [hunk 2]
```diff
@@ -246,15 +246,24 @@ def test_xml_ext_obj(self):
         self.assertEqual(arg.Description, arg2.Description)
         self.assertEqual(arg.DataType, arg2.DataType)
 
+    def test_xml_enum(self):
+        o = self.opc.nodes.objects.add_variable(2, "xmlenum", 0, varianttype=ua.VariantType.Int32, datatype=ua.ObjectIds.ApplicationType)
+        self._test_xml_var_type(o, "enum")
```
Comment (oroulet) [inline tests/tests_xml.py]: Can you add another test for an enum based on EnumValues and not string? us ua.AttributeWriteMask (i=347) for example.

Comment (bitkeeper) [inline tests/tests_xml.py]: test is added

**Comment by oroulet** [comment]
well done!

**Pull request merged**